includes/rest-api/class-sensei-rest-api-messages-controller.php

```diff
@@ -0,0 +1,135 @@
+<?php
+
+if ( ! defined( 'ABSPATH' ) ) {
+	exit;
+} // Exit if accessed directly
+
+/**
+ * A REST controller for the Sensei messages.
+ *
+ * @since 2.3.0
+ *
+ * @see WP_REST_Posts_Controller
+ */
+class Sensei_REST_API_Messages_Controller extends WP_REST_Posts_Controller {
+
+	/**
+	 * Constructor.
+	 *
+	 * @param string $post_type Post type.
+	 */
+	public function __construct( $post_type ) {
+		parent::__construct( $post_type );
+
+		// This filter is needed in order for students to be able to see their own messages only.
+		add_filter( 'rest_sensei_message_query', array( $this, 'exclude_others_comments' ), 10, 2 );
+
+		register_rest_field(
+			'sensei_message',
+			'sender',
+			array(
+				'get_callback' => function( $object ) {
+					return get_post_meta( $object['id'], '_sender', true );
+				},
+				'context'      => array( 'view' ),
+				'schema'       => array(
+					'description' => __( 'The username of the message\'s sender.' ),
+					'type'        => 'string',
+				),
+			)
+		);
+
+		register_rest_field(
+			'sensei_message',
+			'displayed_title',
+			array(
+				'get_callback' => function( $object ) {
+					return Sensei_Messages::get_the_message_title( $object['id'] );
+				},
+				'context'      => array( 'view' ),
+				'schema'       => array(
+					'description' => __( 'The message\'s title.' ),
+					'type'        => 'string',
+				),
+			)
+		);
+
+		register_rest_field(
+			'sensei_message',
+			'displayed_date',
+			array(
+				'get_callback' => function() {
+					return get_the_date();
```

Comment:
I wonder whether we should explicitly pass in the post object here? I guess I'm not sure how the REST API sets up the global context. Does the global `$post` always get set to the current message when this callback is called?

Reply:
I was curious about this my self too. I have tried it with a response which contained multiple messages with different dates and I observed that the dates were correct. So I thought that it calls the_post() or something similar to set the global state correctly. Do you think that we should verify it by code inspection too?